File: src/services/weather.api.ts
export class WeatherService {
  private baseURL = "https://tethys-api-pluviometrico.onrender.com";

  async getRainfall(
    latitude: string | number,
    longitude: string | number
  ): Promise<{ rainfall: number }> {
    const rainfallUrl = `${this.baseURL}/weather/1,${latitude},${longitude}`;
    const rainfallResponse = await fetch(rainfallUrl);
    const json = await rainfallResponse.json();

    if (rainfallResponse.status !== 200) {
      console.log(`Não foi possível encontrar o índice de chuva em ${rainfallUrl}`)
      return { rainfall: 0 }
    }

    const { value: rainfall } = json[0];


    return { rainfall };
  }

  async getLakeVolume() : Promise<{ volume: number }> {
    const random = Math.floor(Math.random() * 100);
    return { volume: random };
  }
}

File: services/weather_api.py
import random
import requests

class WeatherService:
    base_url = "https://tethys-api-pluviometrico.onrender.com"

    def get(self, lat, lon):
        print(f"Buscando dados de chuva para {lat},{lon}")
        url = f"{self.base_url}/weather/1,{lat},{lon}"
        response = requests.get(url)
        result = response.json()

        if result is None or len(result) == 0:
            raise Exception(f"Could not find weather for {lat},{lon}")

        return result[0]["value"]

    def get_lake_volume(self):
        return { "volume": random.randint(0, 100) }
